src/nice-path-decomposition/nice-path-decomposition.h

```diff
@@ -0,0 +1,33 @@
+#include "../centroid-tree.h"
+
+
+class NicePathDecomposition : public PathDecomposition {
+public:
+    enum bag_types {
+        ADD_VERTEX, REMOVE_VERTEX, ADD_EDGE
+    };
+
+    struct Bag {
+        bag_types type;
+        vertex_t v = -1;
+        edge_t edge;
+
+        Bag(bag_types bag_type, vertex_t v) : type(bag_type), v(v) {}
+
+        Bag(bag_types bag_type, edge_t edge) : type(bag_type), edge(edge) {}
+    };
+
+    std::vector<Bag> get_nice_bags();
+
+    NicePathDecomposition(std::vector<std::vector<vertex_t>> &bags, Graph &g);
```

Comment:
Add docstrings for constructors and public methods. Make sure you mention that the bags here are not expected to be nice.